rr_computer_vision/src/lane_detection.cpp

```diff
@@ -20,4 +33,144 @@ LaneDetection::LaneDetection() {
 */
 LaneDetection::~LaneDetection() {
 
+}
+
+void LaneDetection::InitializeSubscribers() {
+    zed_subscriber = nh_.subscribe("/zed/rgb/image_rect_color", 1, &LaneDetection::RGBCameraCallback, this);
+}
+
+void LaneDetection::InitializePublishers() {
+    test_publisher = nh_.advertise<sensor_msgs::Image>("/test_publisher", 1);
+    pointList_pub_ = nh_.advertise<nav_msgs::OccupancyGrid>("/point_vec_out_", 1);
+}
+
+void LaneDetection::RGBCameraCallback(const sensor_msgs::Image& msg){
+
+    // Convert sensor_msgs::Image to a BGR8 cv::Mat
+    cv_bridge::CvImagePtr cv_bridge_bgr = cv_bridge::toCvCopy(msg, sensor_msgs::image_encodings::BGR8);
+    cv::Mat img_bgr8 = cv_bridge_bgr->image;
+    cv::cvtColor(img_bgr8, Im1_HSV_, CV_BGR2HSV, 3);
+
+    //src = (cv::Mat_<float>(4,2) << 500.0, 400.0, 750.0, 400.0, 1280.0, 600.0, 0, 600.0);
+    src = (cv::Mat_<float>(4,2) << 200.0, 400.0, 1150.0, 400.0, 1280.0, 500.0, 20, 580.0);
+    dst = (cv::Mat_<float>(4,2) << 300.0, 0, 900.0, 0, 900.0, 730.0, 300.0, 730.0);
+    
+    M = cv::getPerspectiveTransform(src, dst);
+    cv::warpPerspective(Im1_HSV_,BEV_image,M,img_gray.size());
+
+    multibounds_ = (cv::Mat_<double>(3,6) << 0, 100, 140, 120, 255, 255, 0, 0, 250, 255, 25, 255, 25, 5, 186, 130, 50, 255);
+    Multithreshold(BEV_image, multibounds_, mask_warped_1_);
+
+    bounds_ = cv::Scalar(20, -40);
+    adapt_hsv_patch_size_ = 25;
+    FindWhite(BEV_image, bounds_, adapt_hsv_patch_size_, mask_warped_2_);
+
+    cv::bitwise_or(mask_warped_1_, mask_warped_2_, mask_warped_1_);
+
+    blob_size_ = 100;
+    out = GetContours(mask_warped_1_, blob_size_);
+    
+    cv_bridge::CvImage img_bridge_output;
+    std_msgs::Header header; // empty header
+    header.stamp = ros::Time::now(); // time
+    img_bridge_output = cv_bridge::CvImage(header, sensor_msgs::image_encodings::MONO8, out;
+    
+    // Publish 
+    test_publisher.publish(img_bridge_output.toImageMsg());
+
+    occupancy_.clear();
+    occupancy_.reserve(out.cols * out.rows);
+
+    data_pointer_ = out.data;
+    for (int i = 0; i < (out.rows * out.cols); i++)
+    {
+      value1_ = *data_pointer_;
+      data_pointer_++;
+      if (value1_ == 0)
+      {
+        occupancy_.push_back(-1);
+      }
+      else
+      {
+        occupancy_.push_back(100);
+      }
+    }
+    grid_msg_.data = occupancy_;
+    meta_data_.height = 720;
```

Comment:
Have you confirmed that these parameters will always be constant? Are they impacted by anything?

Reply:
I'm not what those parameters for an occupancy grid represents. I copied them from last year, could you give me some explanations on what they do?